src/components/More.js

```diff
@@ -33,10 +34,29 @@ export class More extends React.Component {
     super(props);
     this.state = {expanded: false};
   }
+
+  // https://discuss.reactjs.org/t/how-to-calculate-jqyery-closest-in-react/1988
+  componentDidMount() {
+    let app = document.getElementById('app');
```

Comment:
That still has the same problem in that you are relying on a global being set outside of this component, which means you are creating an implicit dependency that is difficult to reason about: any code, at any time can change the value of window.__appDiv, causing unexpected behavior.

Frankly, even my (a) suggestion has this problem since it's relying on `document.body` being a global that exists, which means this component could only be rendered inside of a browser environment. If we wanted the component to be isomorphic, and renderable server side _and_ client side, then (b) is the only option really.

Experience has made me paranoid :)